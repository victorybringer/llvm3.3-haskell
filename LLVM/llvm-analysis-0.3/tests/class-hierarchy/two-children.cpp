class Base {
public:
  Base() {}
  virtual void f1() {}
};

class Derived : public Base {
public:
  Derived() : Base() {}
  virtual void f2() {}
};

class Derived2 : public Base {
  int m_i;
public:
  Derived2(int i): m_i(i) {}

  virtual int f6() const { return m_i; }
};

Base * build(int z) {
  if(z)
    return new Derived2(7);

  return new Derived();
}
