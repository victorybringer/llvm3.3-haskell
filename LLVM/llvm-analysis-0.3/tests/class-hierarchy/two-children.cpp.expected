fromList [ ("Base", fromList ["Derived", "Derived2"]) ]
